keel-core-test/src/main/kotlin/com/netflix/spinnaker/keel/persistence/ResourceRepositoryTests.kt

```diff
@@ -255,14 +197,20 @@ abstract class ResourceRepositoryTests<T : ResourceRepository> : JUnit5Minutests
         test("the resource is no longer returned when listing all resources") {
           subject.allResources(callback)
 
-          verify(exactly = 0) { callback(ResourceHeader(resource)) }
+          verify(exactly = 0) { callback(any()) }
         }
 
         test("the resource can no longer be retrieved by name") {
           expectThrows<NoSuchResourceException> {
             subject.get<Map<String, Any>>(resource.metadata.name)
           }
         }
+
+        test("events for the resource are also deleted") {
+          expectThrows<NoSuchResourceException>() {
+            subject.eventHistory(resource.metadata.uid)
+          }
+        }
```

Comment:
Deleting effectively functions as a "stop managing this and clean up" which is pretty useful for testing stuff out in staging.